### ISSUE TITLE
Images in Footer

### ISSUE THREAD

**Issue by bahneclaussen** [bithost-gmbh/pdfviewhelpers#129]
When i insert and pdf:image in pdf:footer the rendering crashes

**Comment by maechler** [comment]
@bahneclaussen I just tested this with our Example from the README, everything seems to work fine with images in the footer. I guess it is a specific problem with your image or setup in general. 

Are there any errors in the PHP error log or the TYPO3 system log?
Could you please share your template?

**output**

![image](https://user-images.githubusercontent.com/4113819/67101737-7d523600-f1c2-11e9-8e91-c310be8a118b.png)

**template**
```html
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers"
      xmlns:pdf="http://typo3.org/ns/Bithost/Pdfviewhelpers/ViewHelpers"
      xsi:schemaLocation="http://typo3.org/ns/Bithost/Pdfviewhelpers/ViewHelpers https://pdfviewhelpers.bithost.ch/schema/2.1.xsd"
      data-namespace-typo3-fluid="true">

<pdf:document outputDestination="inline" title="Bithost Example">
    <pdf:header>
        <pdf:image src="EXT:pdfviewhelpers/Resources/Public/Examples/BasicUsage/logo.png" width="40"/>
        <pdf:text alignment="right" color="#8C8C8C" paragraphSpacing="0" posY="15">
            Bithost GmbH
            Milchbuckstrasse 83
            CH-8057 Zürich

            hallo@bithost.ch
            044 585 28 20

            www.bithost.ch
        </pdf:text>
    </pdf:header>
    <pdf:footer>
        <pdf:graphics.line style="{color: '#8C8C8C'}"/>
        <pdf:image src="EXT:pdfviewhelpers/Resources/Public/Examples/BasicUsage/logo.png" width="20"/>
        <pdf:text alignment="right" color="#8C8C8C" posY="-13.5">EXT:pdfviewhelpers - Basic usage example</pdf:text>
    </pdf:footer>

    <pdf:page>
        <pdf:text posY="50" padding="{bottom: 4}" color="#8C8C8C">
            Zurich, <f:format.date format="d.m.Y">now</f:format.date>
        </pdf:text>
        <pdf:headline>Welcome to the extension pdfviewhelpers</pdf:headline>
        <pdf:text>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr. Sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.</pdf:text>
        <pdf:headline>Some more information</pdf:headline>

        <pdf:multiColumn>
            <pdf:column width="55%">
                <pdf:text>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua:</pdf:text>
                <pdf:list listElements="{0: 'Websites using TYPO3', 1: 'Application Development', 2: 'Mobile Apps', 3: 'Hosting'}"/>
                <pdf:text>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren est Lorem ipsum dolor sit amet.</pdf:text>
            </pdf:column>
            <pdf:column width="45%" padding="{left: 2}">
                <pdf:image src="EXT:pdfviewhelpers/Resources/Public/Examples/BasicUsage/Bithost.jpg"/>
                <pdf:text padding="{top: 1}" color="#8C8C8C">Esteban Marín, Markus Mächler</pdf:text>
            </pdf:column>
        </pdf:multiColumn>

        <pdf:text>Lorem ipsum dolor sit consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.</pdf:text>
        <pdf:text>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.</pdf:text>
    </pdf:page>
</pdf:document>

</html>
```

**Comment by bahneclaussen** [comment]
Hey, 
i have a multisite PDF. 

Code is: 

`<pdf:footer>
            <pdf:text alignment="left" color="{color}" paragraphSpacing="0" posX="14" fontSize="8">{pdf:getPageNumberAlias()} / {pdf:getTotalNumberOfPagesAlias()}   © HOBBY-WOHNWAGENWERK, ING. HARALD STRIEWSKI GMBH</pdf:text>
            <pdf:image src="EXT:nc_hobby_vehicles/Resources/Public/Images/Pdf/hobby-logo.png" width="20"/>
        </pdf:footer>`

Output on first sites is broken and last site works: 

![Bildschirmfoto 2019-10-18 um 16 26 00](https://user-images.githubusercontent.com/45182795/67102654-1df52580-f1c4-11e9-8f50-a21e0e48561d.png)
![Bildschirmfoto 2019-10-18 um 16 26 05](https://user-images.githubusercontent.com/45182795/67102667-22b9d980-f1c4-11e9-862a-78c21fd5fac9.png)

**Comment by maechler** [comment]
Thanks for sharing your template. Indeed, there seems to be a problem on multisite PDFs, but only with images in the footer, the header is not affected.

To me it looks like a bug within TCPDF with PNGs that have an alpha channel, but strangely enough not on all pages. I will have to further investigate this issue. 

To work around this you could try to already pre scale your image the the desired width. In such a way, that you do not have to use `width="20"` anymore.

**Comment by bahneclaussen** [comment]
Hey, 

Scaling has no effects :) 

Thank you for support!

Greetings
Bahne Claussen
Senior Entwickler
 
 
Telefon	 0431 90607-0 <tel:0431 90607-0>
E-Mail	 claussen@new-communication.de <mailto:claussen@new-communication.de>
 
 
 <https://www.new-communication.de/>
 
New Communication GmbH & Co. KG
Werbe- und Marketingagentur
Jägersberg 23  |  24103 Kiel
www.new-communication.de <https://www.new-communication.de/>
 
 
Unseren Marketing- und Trend-Newsletter abonnieren <https://www.new-communication.de/newsletter-bestellen/>
 <https://twitter.com/ncshortservice>	 	 	 <https://www.facebook.com/newcommunication>	 	 	 <https://www.instagram.com/new_communication_kiel/>	 	 	 <https://www.xing.com/companies/newcommunicationgmbh%26co.kg?sc_o=da980_e>
 
Geschäftsführer: Sören Mohr
Handelsregister Kiel HRA 4031 KI
Gerichtsstand Kiel

Komplementärin:
New Communication GmbH & Co. KG
Geschäftsführer: Sören Mohr
Handelsregister Kiel: HRB 5575

This e-mail and any files transmitted with it are confidential and intended solely for the use of the individual or organization to whom they are addressed. Should you not be the intended addressee of this e-mail or his representative, please note that publication, replication of the contents by any means or further communication of the content is not permissible. Should you have received this e-mail in error, please notify the sender.

   <https://app.mailtastic.de/api/li/b6bc88f6-552c-44cc-a477-2122240b5bc0>

**Comment by maechler** [comment]
Rescaling the image worked for me, although the image resolution then is not sufficient anymore. You could also try converting the image to a jpeg or gif. Although I am experiencing orientation issues with these formats, even if I remove EXIF from the images.

**Comment by maechler** [comment]
Another way to workaround this could be to prepare a PDF that only contains the footer. You could do this with word or any other tool and then use this PDF as a template, see https://docs.typo3.org/typo3cms/extensions/pdfviewhelpers/2.1.0/Examples/ExtendExistingPDFs/Index.html.

```html
<pdf:document sourceFile="EXT:pdfviewhelpers/Resources/Public/Examples/ExtendExistingPDFs/pdf_template.pdf">
        <pdf:page importPage="1" margin="{top: 80, right: 20, bottom: 40, left: 20}"> 
       
       </pdf:page>
</pdf:document>
```

**Comment by bahneclaussen** [comment]
If I remove the scaling from the image, it looks the same on all pages up to the last page. Unfortunately I can't use it like this.

**Comment by maechler** [comment]
You could probably also pre scale the image using processing instructions:

`<pdf:image src="EXT:nc_hobby_vehicles/Resources/Public/Images/Pdf/hobby-logo.png" processingInstructions="{width: '60'}" />`

Alternatively you could try converting the image to different image formats like jpg or gif. Or create a template PDF containing only the footer as described in https://github.com/bithost-gmbh/pdfviewhelpers/issues/129#issuecomment-543791029.

**Comment by maechler** [comment]
@bahneclaussen I think I found the underlying issue in TCPDF. There is an issue in the footer if the option `fitonpage` is set for an image: https://github.com/tecnickcom/TCPDF/pull/147

The error should be fixed in `pdfviewhelpers` `v2.2.0`, which is available in TER now. Could you please update the extension and test if this fixes your issue?

If you like the project, please consider giving it a star on GitHub :)

**Comment by maechler** [comment]
@bahneclaussen Could you verify that `v2.2.0` solves your issue?
